modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryFolderModelDocumentContributor.java

```diff
@@ -31,6 +36,42 @@ public void contribute(
 		document.addText(Field.NAME, objectEntryFolder.getName());
 		document.addLocalizedKeyword(
 			"localized_label", objectEntryFolder.getLabelMap(), true, true);
+
+		String[] parts = StringUtil.split(
+			objectEntryFolder.getTreePath(), CharPool.SLASH);
+
+		document.addKeyword(Field.TREE_PATH, parts);
+		document.addKeyword("cms_section", _getCMSSection(parts));
 	}
 
+	private String _getCMSSection(String[] parts) {
+		if (parts.length <= 1) {
+			return "none";
+		}
+
+		ObjectEntryFolder objectEntryFolder =
+			_objectEntryFolderLocalService.fetchObjectEntryFolder(
+				GetterUtil.getLong(parts[1]));
+
+		if (objectEntryFolder == null) {
+			return "none";
+		}
+
+		String externalReferenceCode =
+			objectEntryFolder.getExternalReferenceCode();
+
+		if (externalReferenceCode.equals("L_CONTENT")) {
+			return "content";
```

Comment:
@adolfopa, just a question, wouldn't it be better to have a constants class for this? Or is is mostly done to hide this feature?

Reply:
I agree 100%, but as there's only one place it is used (here), we cannot introduce the constant :-(